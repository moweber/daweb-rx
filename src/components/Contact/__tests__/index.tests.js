import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ContactForm from "..";

afterEach(cleanup);

describe("Contact component renders", () => {
  it("renders", () => {
    render(<ContactForm />);
  });

  it("matches snapshot DOM node structure", () => {
    const { asFragment } = render(<ContactForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});

it("inserts text into form button", () => {
  const { getByTestId } = render(<ContactForm />);
  expect(getByTestId("button")).toHaveTextContent("Submit");
});
