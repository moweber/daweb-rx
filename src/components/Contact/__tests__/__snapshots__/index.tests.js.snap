// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component renders matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <section
    class="container mContact"
  >
    <form
      id="contact-form"
    >
      <div
        class="form-group"
      >
        <div>
          <label
            for="name"
          >
            Name:
          </label>
          <input
            class="form-control"
            name="name"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="email"
          >
            Email address:
          </label>
          <input
            class="form-control"
            name="email"
            type="email"
            value=""
          />
        </div>
        <div>
          <label
            for="message"
          >
            Message:
          </label>
          <textarea
            class="form-control"
            name="message"
            rows="5"
          />
        </div>
        <button
          class="btn btn-primary"
          data-testid="button"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </section>
</DocumentFragment>
`;
